third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -139,126 +117,90 @@ bool dds_device::impl::init()
-                        std::string sensor_name = stream_name;                         // todo
-                        auto & stream = _streams[stream_name];
-                        if( stream )
-                            DDS_THROW( runtime_error, "stream '" + stream_name + "' already exists" );
-                        stream = std::make_shared< dds_motion_stream >( stream_name, sensor_name );
-                        dds_stream_profiles profiles;
-                        int default_profile_index = 0;
-                        for( int i = 0; i < motion_stream_profiles->num_of_profiles; ++i )
+                        for( auto profile : j["profiles"] )
                         {
-                            auto const & profile = motion_stream_profiles->profiles[i];
-                            profiles.push_back( to_realdds_profile( profile ) );
-                            if( profile.default_profile )
-                                default_profile_index = i;
+                            auto frequency = utilities::json::get< int16_t >( profile, "frequency" );
+                            dds_stream_format format( utilities::json::get< std::string >( profile, "format" ) );
+                            profiles.push_back( std::make_shared< dds_motion_stream_profile >(
+                                dds_stream_uid( (uint32_t)_streams.size() ),
+                                format,
+                                frequency ) );
                         }
-                        stream->init_profiles( profiles, default_profile_index );
-                        LOG_INFO( "... MOTION_STREAM_PROFILES: " << _streams.size() << "/" << _expected_num_of_streams
-                                                                 << " streams received" );
-                        if( _streams.size() >= _expected_num_of_streams )
-                            state = state_type::DONE;
                     }
                     else
-                        LOG_ERROR( "... MOTION_STREAM_PROFILES unexpected" );
-                    break;
-
-                default:
-                    LOG_ERROR( "... Wrong message (" << (int)data._msg_type << ") received" );
-                    break;
+                        DDS_THROW( runtime_error, "stream '" + stream_name + "' is of unknown type '" + type + "'" );
+                    if( default_profile_index < 0 || default_profile_index >= profiles.size() )
+                        DDS_THROW( runtime_error,
+                                   "stream '" + stream_name + "' default profile index "
+                                       + std::to_string( default_profile_index ) + " is out of bounds" );
+
+                    auto & stream = _streams[stream_name];
+                    if( stream )
+                        DDS_THROW( runtime_error, "stream '" + stream_name + "' already exists" );
+                    stream = std::make_shared< dds_video_stream >( stream_name, sensor_name );
```

Comment:
Hard to see in this UI
Where is `dds_motion_stream` created?

Reply:
This was a bug. Once we have it working we need to add some unit-tests with motion streams...

Reply:
Added a test-case.
Moved profile `type_to_string()` into stream-base `type_string()`.
Added type-check: when creating a stream in device-impl, it checks that its type is now the same as that given in the json.